apps/emqx_mq/mix.exs

```diff
@@ -0,0 +1,44 @@
+defmodule EMQXMQ.MixProject do
+  use Mix.Project
+  alias EMQXUmbrella.MixProject, as: UMP
+
+  def project do
+    [
+      app: :emqx_mq,
+      version: "6.0.0",
+      build_path: "../../_build",
+      compilers: [:elixir, :asn1, :erlang, :app],
+      erlc_options: UMP.erlc_options(),
+      erlc_paths: ["gen_src" | UMP.erlc_paths()],
+      # used by our `compile.asn1` compiler
+      asn1_srcs: asn1_srcs(),
+      deps_path: "../../deps",
+      lockfile: "../../mix.lock",
+      elixir: "~> 1.14",
+      start_permanent: Mix.env() == :prod,
+      deps: deps()
+    ]
+  end
+
+  def asn1_srcs() do
+    "./asn.1/*.asn"
+    |> Path.wildcard()
+    |> Enum.map(fn src ->
+      %{src: src, compile_opts: [:per, :noobj, outdir: ~c"gen_src"]}
```

Comment:
1. I don't think per is a good option. It's not extendable.
2. I prefer when each schema is listed individually, rather than using wildcards and map. This way it's harder to accidentally change compile options for all schemas and make old data unreadable.